borsh-derive/src/internals/attributes/field/mod.rs

```diff
@@ -115,7 +120,16 @@ impl From<BTreeMap<Symbol, Variants>> for Attributes {
 }
 
 pub(crate) fn contains_skip(attrs: &[Attribute]) -> bool {
```

Comment:
Start with:
```bash
git apply skip.diff
```
skip.diff
```diff
diff --git a/borsh-derive/src/internals/attributes/field/mod.rs b/borsh-derive/src/internals/attributes/field/mod.rs
index 485561b8..182b5688 100644
--- a/borsh-derive/src/internals/attributes/field/mod.rs
+++ b/borsh-derive/src/internals/attributes/field/mod.rs
@@ -1,36 +1,37 @@
 use std::collections::BTreeMap;
 
 use once_cell::sync::Lazy;
 use syn::{meta::ParseNestedMeta, Attribute, WherePredicate};
 
 use self::bounds::BOUNDS_FIELD_PARSE_MAP;
 
 use super::{
     parsing::{attr_get_by_symbol_keys, meta_get_by_symbol_keys, parse_lit_into},
-    BoundType, Symbol, BORSH, BOUND, DESERIALIZE_WITH, SERIALIZE_WITH, SKIP,
+    BoundType, Symbol, BORSH, BOUND, DESERIALIZE_WITH, SERIALIZE_WITH, SKIP, SKIP_SMALL,
 };
 
 #[cfg(feature = "schema")]
 use {
     super::schema_keys::{PARAMS, SCHEMA, WITH_FUNCS},
     schema::SCHEMA_FIELD_PARSE_MAP,
 };
 
 pub mod bounds;
 #[cfg(feature = "schema")]
 pub mod schema;
 
 enum Variants {
     Bounds(bounds::Bounds),
     SerializeWith(syn::ExprPath),
     DeserializeWith(syn::ExprPath),
+    Skip(()),
     #[cfg(feature = "schema")]
     Schema(schema::Attributes),
 }
 
 type ParseFn = dyn Fn(Symbol, Symbol, &ParseNestedMeta) -> syn::Result<Variants> + Send + Sync;
 
 static BORSH_FIELD_PARSE_MAP: Lazy<BTreeMap<Symbol, Box<ParseFn>>> = Lazy::new(|| {
     let mut m = BTreeMap::new();
     // assigning closure `let f = |args| {...};` and boxing closure `let f: Box<ParseFn> = Box::new(f);`
     // on 2 separate lines doesn't work
@@ -50,71 +51,84 @@ static BORSH_FIELD_PARSE_MAP: Lazy<BTreeMap<Symbol, Box<ParseFn>>> = Lazy::new(|
             .map(Variants::DeserializeWith)
     });
 
     #[cfg(feature = "schema")]
     let f_schema: Box<ParseFn> = Box::new(|_attr_name, _meta_item_name, meta| {
         let map_result = meta_get_by_symbol_keys(SCHEMA, meta, &SCHEMA_FIELD_PARSE_MAP)?;
         let schema_attributes: schema::Attributes = map_result.into();
         Ok(Variants::Schema(schema_attributes))
     });
 
+    let f_skip: Box<ParseFn> = Box::new(|attr_name, meta_item_name, meta| {
+        Ok(Variants::Skip(()))
+    });
+
     m.insert(BOUND, f_bounds);
     m.insert(SERIALIZE_WITH, f_serialize_with);
     m.insert(DESERIALIZE_WITH, f_deserialize_with);
+    m.insert(SKIP_SMALL, f_skip);
     #[cfg(feature = "schema")]
     m.insert(SCHEMA, f_schema);
     m
 });
 
 #[derive(Default, Clone)]
 pub(crate) struct Attributes {
     pub bounds: Option<bounds::Bounds>,
     pub serialize_with: Option<syn::ExprPath>,
     pub deserialize_with: Option<syn::ExprPath>,
+    pub skip: bool,
     #[cfg(feature = "schema")]
     pub schema: Option<schema::Attributes>,
 }
 
 impl From<BTreeMap<Symbol, Variants>> for Attributes {
     fn from(mut map: BTreeMap<Symbol, Variants>) -> Self {
         let bounds = map.remove(&BOUND);
         let serialize_with = map.remove(&SERIALIZE_WITH);
         let deserialize_with = map.remove(&DESERIALIZE_WITH);
+        let skip = map.remove(&SKIP_SMALL);
         let bounds = bounds.map(|variant| match variant {
             Variants::Bounds(bounds) => bounds,
             _ => unreachable!("only one enum variant is expected to correspond to given map key"),
         });
 
         let serialize_with = serialize_with.map(|variant| match variant {
             Variants::SerializeWith(serialize_with) => serialize_with,
             _ => unreachable!("only one enum variant is expected to correspond to given map key"),
         });
 
         let deserialize_with = deserialize_with.map(|variant| match variant {
             Variants::DeserializeWith(deserialize_with) => deserialize_with,
             _ => unreachable!("only one enum variant is expected to correspond to given map key"),
         });
 
+        let skip = skip.map(|variant| match variant {
+            Variants::Skip(skip) => skip,
+            _ => unreachable!("only one enum variant is expected to correspond to given map key"),
+        });
+
         #[cfg(feature = "schema")]
         let schema = {
             let schema = map.remove(&SCHEMA);
             schema.map(|variant| match variant {
                 Variants::Schema(schema) => schema,
                 _ => {
                     unreachable!("only one enum variant is expected to correspond to given map key")
                 }
             })
         };
         Self {
             bounds,
             serialize_with,
             deserialize_with,
+            skip: skip.is_some(),
             #[cfg(feature = "schema")]
             schema,
         }
     }
 }
 
 pub(crate) fn contains_skip(attrs: &[Attribute]) -> bool {
     attrs.iter().any(|attr| attr.path() == SKIP)
 }
 
diff --git a/borsh-derive/src/internals/attributes/mod.rs b/borsh-derive/src/internals/attributes/mod.rs
index b0f90168..388a8f71 100644
--- a/borsh-derive/src/internals/attributes/mod.rs
+++ b/borsh-derive/src/internals/attributes/mod.rs
@@ -14,20 +14,21 @@ pub const BORSH: Symbol = Symbol("borsh", "borsh(...)");
 /// bound - sub-borsh nested meta, field-level only, `BorshSerialize` and `BorshDeserialize` contexts
 pub const BOUND: Symbol = Symbol("bound", "bound(...)");
 //  use_discriminant - sub-borsh nested meta, item-level only, enums only, `BorshSerialize` and `BorshDeserialize` contexts
 pub const USE_DISCRIMINANT: Symbol = Symbol("use_discriminant", "use_discriminant = ...");
 /// serialize - sub-bound nested meta attribute
 pub const SERIALIZE: Symbol = Symbol("serialize", "serialize = ...");
 /// deserialize - sub-bound nested meta attribute
 pub const DESERIALIZE: Symbol = Symbol("deserialize", "deserialize = ...");
 /// borsh_skip - field-level only attribute, `BorshSerialize`, `BorshDeserialize`, `BorshSchema` contexts
 pub const SKIP: Symbol = Symbol("borsh_skip", "borsh_skip");
+pub const SKIP_SMALL: Symbol = Symbol("skip", "skip");
 /// init - sub-borsh nested meta, item-level only attribute  `BorshDeserialize` context
 pub const INIT: Symbol = Symbol("init", "init = ...");
 /// serialize_with - sub-borsh nested meta, field-level only, `BorshSerialize` context
 pub const SERIALIZE_WITH: Symbol = Symbol("serialize_with", "serialize_with = ...");
 /// deserialize_with - sub-borsh nested meta, field-level only, `BorshDeserialize` context
 pub const DESERIALIZE_WITH: Symbol = Symbol("deserialize_with", "deserialize_with = ...");
 
 #[cfg(feature = "schema")]
 pub mod schema_keys {
     use super::Symbol;
```

Checklist: 
- [ ] `git apply skip.diff`
- [ ] remove `pub(crate) fn contains_skip(attrs: &[Attribute]) -> bool ` completely
- [ ] change signature of `Attributes::parse`: `pub(crate) fn parse(attrs: &[Attribute], skipped: bool) -> Result<Self, syn::Error>`  => `pub(crate) fn parse(attrs: &[Attribute]) -> Result<Self, syn::Error>`